### DISCUSSION TITLE
Custom Detekt Rule Question

### DISCUSSION THREAD

**Discussion by th-jtullgren** [detekt/detekt#4582] [Q&A]
Hi,

I am looking to create a rule to check for duplicate uses of integers for subtypes of a specific interface. Example:

```
interface MyInterface { val anInt: Int }
enum class MyEnumA(override val anInt: Int) { EnumVal(1); }
enum class MyEnumB(override val anInt: Int) { EnumVal(1); }
```

In the above example I want to log a code smell saying that MyEnumB duplicates the anInt value by using 1 again.

I am not very familiar with Psi and could not readily figure out which visitor method(s) could provide me the data I need to check. I have written this rule using the Android API Lint but I want to stick with just Detekt if possible. 

Example of the Android Api Lint Rule:

```
    private class DuplicateSystemCodeElementHandler(private val context: JavaContext): UElementHandler() {
        override fun visitEnumConstant(node: UEnumConstant) {
            node.type.superTypes.firstOrNull { it.canonicalText == "com.company.SystemCode" }
                ?: return
            // check value
        }
    }
```

Any help would be appreciated on which methods in the Rule api I need to use.

Thanks!

**Comment by cortinico** [comment]
> I am looking to create a rule to check for duplicate uses of integers for subtypes of a specific interface. Example:
> 
> ```
> interface MyInterface { val anInt: Int }
> enum class MyEnumA(override val anInt: Int) { EnumVal(1); }
> enum class MyEnumB(override val anInt: Int) { EnumVal(1); }
> ```
> 
> In the above example I want to log a code smell saying that MyEnumB duplicates the anInt value by using 1 again.

I'm failing to understand this rule. Could you please provide more examples + some real-world use case for this?
